ui/apps/nft-portal/src/flow/utils.tsx

```diff
@@ -352,6 +352,37 @@ export async function proposeNFTToCatalog(
   }
 
   const cadence = catalogJson.transactions.propose_nft_to_catalog
+
+
+  // Check if the image is an IPFS one, and convert it to one that is 
+  // easily viewable from the catalog.
+  let collectionSquareImage = sampleNFTView.NFTCollectionDisplay.collectionSquareImage.file.url
+  if (sampleNFTView.NFTCollectionDisplay.collectionSquareImage.file.cid) {
+    const cid = sampleNFTView.NFTCollectionDisplay.collectionSquareImage.file.cid
+    const path = sampleNFTView.NFTCollectionDisplay.collectionSquareImage.file.path || ""
+    // Using pinata public gateway, since we don't have a proper ipfs gateway as part of the view.
```

Comment:
This seems pretty bad right? Should we open an issue in flow-nft to support this?

Reply:
Yeah, I think maybe proposing to add an optional `public gateway` to the ipfs file view would be really helpful.

I think we should get this in regardless, since that might take a while. Alternatively we could store just the IPFS contents rather than a public pinata link - but i went with this since it would make consuming the catalog a bit easier and not require others to randomly choose some ipfs link.